documentation/docs/assertions/eventually.md

```diff
@@ -0,0 +1,129 @@
+---
+id: eventually
+title: Eventually
+slug: eventually.html
+---
+
+
+When testing non-deterministic code, a common use case is "I expect this code to pass after a short period of time".
+
+For example, if you were testing a IO operation, you might need to wait until the IO operation has flushed.
+
+Sometimes you can do a Thread.sleep but this is isn't ideal as you need to set a sleep threshold high enough so that it
+won't expire prematurely on a slow machine. Plus it means that your test will sit around waiting on the timeout even if
+the code completes quickly on a fast machine.
+
+Or you can roll a loop and sleep and retry and sleep and retry, but this is just boilerplate slowing you down.
+
+Another common approach is to use countdown latches and this works fine if you are able to inject the latches in the
+appropriate places but it isn't always possible to have the code under test trigger a latch.
+
+As an alternative, kotest provides the `eventually` function and the `Eventually` configuration which periodically test
+the code ignoring your specified exceptions and ensuring the result satisfies an optional predicate, until the timeout
+is eventually reached or too many iterations have passed. This is flexible and is perfect for testing nondeterministic
+code.
+
+
+### Examples
+
+
+#### Simple examples
+
+Let's assume that we send a message to an asynchronous service.
+After the message is processed, a new row is inserted into user table.
+
+We can check this behaviour with our `eventually` function.
+
+```kotlin
+class MyTests : ShouldSpec() {
+  init {
+    should("check if user repository has one row after message is sent") {
+      sendMessage()
+      eventually(5.seconds) {
+        userRepository.size() shouldBe 1
+      }
+    }
+  }
+}
+```
+
+#### Exceptions
+
+By default, `eventually` will ignore any exception that is thrown inside the function (note, that means it won't catch `Error`).
```

Comment:
I think this will only ignore `AssertionError`s thrown by default

Reply:
Yes good catch.